interceptors/logging/interceptors.go

```diff
@@ -47,40 +50,102 @@ func (c *reporter) PostCall(err error, duration time.Duration) {
 	if err != nil {
 		logger = logger.With("error", fmt.Sprintf("%v", err))
 	}
-	logger.With(c.opts.durationFieldFunc(duration)...).Log(c.opts.levelFunc(code), fmt.Sprintf("finished %s %s call", c.kind, c.typ))
+
+	if !c.opts.shouldLogRequest(interceptors.FullMethod(c.service, c.method), err) {
```

Comment:
I don't get this, why we need this `if`? I think `shoullLog` controls this :thinking:

Reply:
I have added another decider to check if request logging is enabled.
My previous decider was breaking the tests and the initial API, so I thought that by using the decider as an option instead of my previous implementation, would break no changes, hence one more condition

Reply:
Hm, think about what is the user experience here.

"If I configure `to log` I will have `finished .,.. call` log line."
"If I configure `to log request` I will have `request finished` if server side and `response finished` on caller. "
"If I configure `to log request, but to not log` I will have no log line"

Do you those three perceptions of current implementation will be ok for user? Because I think it can confuse users a lot. I think we can improve it much more :muscle: Think about how would configure this and what do you want to achieve. To me It's not `shouldLogRequest`. Focus on what you change here. Because to me, nothing is really needed. Why not just leaving `finished .. call`? (: 

What request finished` means and how `response` can be finished? `response` is  a noun?

Reply:
> Do you those three perceptions of current implementation will be ok for user?

@bwplotka The confusion that arises out of the three implementation sounds fair. I think due to this confusion, it would make sense to allow one decider, rather than two currently in this PR

> Because to me, nothing is really needed. Why not just leaving finished .. call? (:

So should we skip out this PR altogether, because if the changes don't make sense and doesn't achieve as we wanted, it would make sense to skip out this PR 🙂

Reply:
Currently, I configured the grpc middleware in Thanos, and the logs currently look like this before this pr gets merged - 
```
level=info name=sidecar-0 ts=2020-07-20T08:48:49.680407947Z caller=logger.go:31 service=gRPC/server component=sidecar system=grpc kind=server grpc.service=thanos.Store grpc.method=Info grpc.method_type=unary grpc.start_time=2020-07-20T14:18:49+05:30 grpc.request.deadline=2020-07-20T14:18:54+05:30 peer.address=127.0.0.1:57518 grpc.request.request-id=01EDNPJK0G6ZC1YHWNCVDMYVSN grpc.code=OK grpc.time_ms=0.054 finishedserverunarycall=(MISSING)
level=info name=sidecar-2 ts=2020-07-20T08:48:49.680587108Z caller=logger.go:31 service=gRPC/server component=sidecar system=grpc kind=server grpc.service=thanos.Store grpc.method=Info grpc.method_type=unary grpc.start_time=2020-07-20T14:18:49+05:30 grpc.request.deadline=2020-07-20T14:18:54+05:30 peer.address=127.0.0.1:37034 grpc.request.request-id=01EDNPJK0GX54BY58ACJ2XSZXW grpc.code=OK grpc.time_ms=0.044 finishedserverunarycall=(MISSING)
level=info name=sidecar-1 ts=2020-07-20T08:48:49.68244212Z caller=logger.go:31 service=gRPC/server component=sidecar system=grpc kind=server grpc.service=thanos.Store grpc.method=Info grpc.method_type=unary grpc.start_time=2020-07-20T14:18:49+05:30 grpc.request.deadline=2020-07-20T14:18:54+05:30 peer.address=127.0.0.1:59918 grpc.request.request-id=01EDNPJK0G94RMSP7CPEEB8J2D grpc.code=OK grpc.time_ms=0.034 finishedserverunarycall=(MISSING)
```

If the metadata required here looks comprehensive, then we can altogether skip merging this PR.